From 67d7a08f00f1defafa4d79f33f1f8aa4d726a907 Mon Sep 17 00:00:00 2001
From: krp97 <k.plata576@gmail.com>
Date: Thu, 22 Jul 2021 14:40:31 +0000
Subject: [PATCH 3/5] Remove condition for pausing/resuming the STC

This commit adjusts the condition, which prevented audio / video
stc_channel from being paused on calls invoked from the sinks.

The internal state of the boolean variable is tracked based on calls
made explicitly through the system clock. The underlying function
is called in different brcm plugins, which leads to inconsistencies
with the value.
---
 reference/util/src/gst_system_clock.c | 18 ++++++------------
 reference/util/src/gst_system_clock.h |  1 -
 2 files changed, 6 insertions(+), 13 deletions(-)

diff --git a/reference/util/src/gst_system_clock.c b/reference/util/src/gst_system_clock.c
index a3f1696..117307e 100755
--- a/reference/util/src/gst_system_clock.c
+++ b/reference/util/src/gst_system_clock.c
@@ -182,20 +182,14 @@ void gst_brcm_system_clock_freeze_stc_channel(NEXUS_SimpleStcChannelHandle stc_c
         BDBG_WRN(("Stc channel not found"));
     } else {
         if (freeze) {
-            if (!sysclock.clock[token_id].frozen[SYS_CLK_CLIENT_VIDEO] && !sysclock.clock[token_id].frozen[SYS_CLK_CLIENT_AUDIO]){
-                NEXUS_Error rc = NEXUS_SimpleStcChannel_Freeze(stc_channel, true);
-                if (rc) {
-                    BDBG_WRN(("NEXUS_SimpleStcChannel_Freeze failed %d", GST_FUNCTION,__LINE__, rc));
-                }
+            NEXUS_Error rc = NEXUS_SimpleStcChannel_Freeze(stc_channel, true);
+            if (rc) {
+                BDBG_WRN(("NEXUS_SimpleStcChannel_Freeze failed %d", GST_FUNCTION,__LINE__, rc));
             }
-            sysclock.clock[token_id].frozen[sys_clk_client] = true;
         } else { /* freeze == 0 */
-            sysclock.clock[token_id].frozen[sys_clk_client] = false;
-            if (!sysclock.clock[token_id].frozen[SYS_CLK_CLIENT_VIDEO] && !sysclock.clock[token_id].frozen[SYS_CLK_CLIENT_AUDIO]){
-                NEXUS_Error rc = NEXUS_SimpleStcChannel_Freeze(stc_channel, false);
-                if (rc) {
-                    BDBG_WRN(("NEXUS_SimpleStcChannel_Freeze failed %d", GST_FUNCTION,__LINE__, rc));
-                }
+            NEXUS_Error rc = NEXUS_SimpleStcChannel_Freeze(stc_channel, false);
+            if (rc) {
+                BDBG_WRN(("NEXUS_SimpleStcChannel_Freeze failed %d", GST_FUNCTION,__LINE__, rc));
             }
         }
     }
diff --git a/reference/util/src/gst_system_clock.h b/reference/util/src/gst_system_clock.h
index 6486921..54611d2 100644
--- a/reference/util/src/gst_system_clock.h
+++ b/reference/util/src/gst_system_clock.h
@@ -60,7 +60,6 @@ typedef struct SystemClock {
     NEXUS_SimpleStcChannelHandle stc_channel;
     unsigned int token;
     unsigned int ref_count;
-    bool frozen[SYS_CLK_CLIENT_MAX];
 }Clock;
 
 struct _GstBrcmSystemClock {
-- 
2.7.4

